bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/ExtendedDiscoveryServiceMock.groovy

```diff
@@ -35,6 +37,11 @@ class ExtendedDiscoveryServiceMock extends DiscoveryServiceMock implements Exten
 
     @Override
     public void startScan() {
-        thingDiscovered(new DiscoveryResultImpl(new ThingUID(thingType, "foo"), null, null, null, null, DEFAULT_TTL))
+        thingDiscovered(new DiscoveryResultImpl(new ThingUID(thingType, "foo"), null, this.discoveryProperties, null, null, DEFAULT_TTL))
     }
+    
+    public void setDiscoveryProperties(Map<String, Object> _discoveryProperties) {
+        this.discoveryProperties = _discoveryProperties;
```

Comment:
we generally don't use c++ best practices here. Simply name it `discoveryProperties`